src/main.ts

```diff
@@ -0,0 +1,52 @@
+import { Actor } from 'apify';
+import { PlaywrightCrawler } from 'crawlee';
+
+import { LABELS } from './consts.js';
+import { router } from './router.js';
+import actorState from './actor-state.js';
+
+interface Input {
+    searchPhrase: string;
+    category: string;
+    maxProducts?: number;
+    scrapeReviews: boolean;
+    maxReviews?: number
+}
+
+await Actor.init();
+
+const input = await Actor.getInput<Input>() ?? {} as Input;
+
+const proxyConfiguration = await Actor.createProxyConfiguration({
+    groups: ['RESIDENTIAL'],
```

Comment:
Did you get too many blocks with datacenter? Pls write a comment why you chose RESIDENTIAL, how much it was blocking before. First you should try to increase `maxRequestRetries` a bit to like 5 or even 10.